rllib/evaluation/per_policy_sample_collector.py

```diff
@@ -0,0 +1,487 @@
+import logging
+import numpy as np
+from typing import Dict, Optional
+
+from ray.rllib.evaluation.episode import MultiAgentEpisode
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.policy.view_requirement import ViewRequirement
+from ray.rllib.utils.framework import try_import_tf, try_import_torch
+from ray.rllib.utils.types import AgentID, EnvID, EpisodeID, TensorType
+
+tf1, tf, tfv = try_import_tf()
+torch, _ = try_import_torch()
+
+logger = logging.getLogger(__name__)
+
+
+class _PerPolicySampleCollector:
```

Comment:
It's a bit confusing this is called a sample collector but doesn't inherit the interface (presumably since it doesn't have policy id). Should it be called something slightly different?